release-application.sh

```diff
@@ -0,0 +1,167 @@
+    BUILD_PROFILE=${PROPERTY_VALUE}
+}
+
+extract_application_versions() {
+
+    # Extract the current application version
+    local CURRENT_VERSION=`printf 'APPLICATION_VERSION=${project.version}\n0\n' |\
+        mvn help:evaluate |\
+        grep '^APPLICATION_VERSION=' |\
+        sed -r 's/(^APPLICATION_VERSION=)(.+)(-SNAPSHOT|-RELEASE)/\2/'`
+
+    RELEASED_VERSION=${CURRENT_VERSION}-RELEASE
+
+    # Extract the versions
+    local MAJOR=`echo ${CURRENT_VERSION} | sed -r 's/([0-9]+).([0-9]+).([0-9]+)/\1/'`
+    local MINOR=`echo ${CURRENT_VERSION} | sed -r 's/([0-9]+).([0-9]+).([0-9]+)/\2/'`
+    local PATCH=`echo ${CURRENT_VERSION} | sed -r 's/([0-9]+).([0-9]+).([0-9]+)/\3/'`
+
+    if [ ${HOTFIX} = 1 ]
+    then
+        RELEASED_VERSION="${MAJOR}.${MINOR}.$((PATCH+1))-RELEASE"
+    else
+        NEXT_VERSION="${MAJOR}.$((MINOR+1)).0-SNAPSHOT"
+    fi
+}
+
+update_version_in_application_properties() {
+    if [ -f ${APPLICATION_PROPERTIES} ]
+    then
+        sed -i.bak -r "s/^${APPLICATION_VERSION_PROPERTY}\s*=\s*.+/${APPLICATION_VERSION_PROPERTY} = $1/" ${APPLICATION_PROPERTIES}
+        rm -f ${APPLICATION_PROPERTIES}.bak
+        git add ${APPLICATION_PROPERTIES}
+    fi
+}
+
+update_current_version() {
+    local VERSION=$1
+    mvn versions:set -DnewVersion=${VERSION}
+    git add pom.xml
+
+    update_version_in_application_properties ${VERSION}
+}
+
+init_script_variables() {
+
+    extract_release_branch_var
+
+    extract_is_hotfix_var
+
+    extract_development_branch_var
+
+    extract_build_profile_var
+}
+
+# Build the release - we will make sure the tests are passing and the build is immutable and repeatable (no snapshots).
+build_release_artifact() {
+
+    # First merge the current development branch into release branch
+    git checkout ${RELEASE_BRANCH_NAME}
+    git merge --no-ff origin/${DEVELOPMENT_BRANCH_NAME} -m "M"
+
+    extract_application_versions
+
+    update_current_version ${RELEASED_VERSION}
+
+    git commit --amend -m "Released version: ${RELEASED_VERSION}."
+
+    # Perform the build.
+    mvn clean install -P${BUILD_PROFILE} -DsnapshotsAllowed=false
+}
+
+# Deploy the released artifact to Nexus
+deploy_artifact() {
+    # Skip tests - they were already ran in the build step
+    mvn deploy -P${BUILD_PROFILE} -DskipTests
+}
+
+# Update the scm state according to our release workflow.
+prepare_next_iteration() {
+
+    # Push the changes to remote.
+    git push origin "${RELEASE_BRANCH_NAME}"
+
+    # Create a tag of the current release so we will be able to easily perform hotfixes if needed from this point.
+    mvn scm:tag -Dtag="v${RELEASED_VERSION}"
+
+    # For hotfixes we won't do anything after the hotfix branch  was merged in the release.
+    if [ ${HOTFIX} = 0 ]
+    then
+        # Move the current development branch on top of the current release checkpoint.
+        git checkout ${DEVELOPMENT_BRANCH_NAME}
+        git rebase "v${RELEASED_VERSION}"
```

Comment:
Why are we rebasing `develop` here? All code changes should be merged via PR process...